src/src/SelfServiceApiClient.js

```diff
@@ -130,6 +130,19 @@ export class SelfServiceApiClient {
     return data.items;
   }
 
+  async validateMessageSchema(topicId, messageContract) {
+    const response = await this.requestWithToken(
+      composeUrl(`kafkatopics/${topicId}/messagecontracts-validate`),
+      "POST",
+      messageContract,
+    );
+    if (response.detail !== "") {
+      return response;
+    }
+    let obj = await response.json();
+    return obj || [];
+  }
+
```

Comment:
Make a validation object here instead with fields:
IsValid: bool,
FailureReason: string

and use this object to encapsulate the API call so that the outside dont know about the details